helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -321,8 +321,13 @@ public Response updateCluster(@PathParam("clusterId") String clusterId,
           customFieldsMap =
               OBJECT_MAPPER.readValue(content, new TypeReference<HashMap<String, String>>() {
               });
-          // content is given as a KV mapping. Nullify content
+          // content is given as a KV mapping. Nullify content unless (case-insensitive) reason key present in map
           content = null;
+          for (Map.Entry<String, String> entry : customFieldsMap.entrySet()) {
```

Comment:
just wondering: is  the fast way to look if the key exists? but if  we write as "reason" vs/vs "ReAsoN" then that approach won't work.

Reply:
As far as I know this is the best solution to find a key case insensitive key without making changes to the map data structure, or using something a treemap like this:
TreeMap<>(String.CASE_INSENSITIVE_ORDER);

Fortunately the map should be fairly small from what I've seen, 3-10 entries, so this inefficiency shouldn't have an impact on overall performance of the call

Reply:
good to know.. thanks for working on this.